### ISSUE TITLE
Hangs when extracting a squashfs file from an ISO

### ISSUE THREAD

**Issue by JesseTG** [unetbootin/unetbootin#50]
I'm using Ubuntu 15.04, and have tried installing both Kubuntu 15.10 and Parted Magic (forget the version) to a 2GB flash drive.  UNetbootin does not crash or otherwise deliver errors or warnings, but I'm seeing lots of I/O without seemingly any progress...but *only* when it extracts a squashfs file from the original ISO.

**Comment by gkovacs** [comment]
no, there's nothing wrong. the squashfs file is by far the largest file on the ISO, so it obviously will take the longest time to extract. progress is indicated in terms of percentage of files extracted at the moment, not percentage of data written)

**Comment by E3V3A** [comment]
Yes, this is a problem, since it doesn't show any kind of progress. I suggest a possible improvement by displaying the status of the pipe (or `dd`) if this is what is done. What is the exact command used, that extracts the squashfs?

**Comment by gastoneb** [comment]
I had a similar issue. I ran UNetBootin and killed it when I thought the program was hung. Then I tried to process/transfer the iso again and it completed successfully in a short amount of time. Great! except that the job was in no way complete. I had to reformat the disk and try again, with more patience. 

It's as if UNetBootin looked for a particular file, saw there was something with the correct name, and moved to the next task without checking that the file was actually valid.

**Comment by spbond** [comment]
I would just like to point out that I am having this problem, and unetbootin reports the size of filesystem.squashfs to be 1,416 MB.  The size of filesystem.squashfs on the usb drive I am copying to is 1,485 MB and not growing.  To me this appears to be hung, perhaps on the file which actually comes after filesystem.squashfs?  Just to be sure, I let it continue running for an inordinate amount of time, something like 20 to 30 minutes, with no change.

**Comment by gkovacs** [comment]
No it's not stuck. It pre-allocated the file hence why it reports the full
size, but needs to actually flush and let the contents finish writing to
disk. Just wait and let the extraction finish.

ᐧ

**Comment by thelastinuit** [comment]
In mac, the Activity Monitor says not responding and I have left it run more than an hour, JIC, but nothing... is someone else having a similar issue?

**Comment by philipstancil** [comment]
@thelastinuit I'm having the same issue. I left mine running for over six hours. It was stuck. I've repeated the process over five different times, and it always blows past the file it got stuck on, but then gets stuck on something else.

@gkovacs Thanks for investing your time into building this tool! Can you provide any idea as to how to move past the extraction phase? I'm running macOS Sierra (10.12.4), and I'm trying to install from an ISO, not a distribution. The ISO is Win10_1607_English_x64.iso, and I got it straight from Microsoft's website. Currently the app is hung at extracting file 78 of 985 files.

Thanks!

**Comment by thelastinuit** [comment]
@philipstancil I've noticed that not all ISO do this stuck thingy. The options at the drop down menu work just fine. Weird.

**Comment by philipstancil** [comment]
@thelastinuit Gotcha. I looked in that distributions dropdown to see if I could use it to get Windows directly, but no such luck. I imagine that's due to licensing restrictions of some kind.

Thanks for the thoughts, Luis!

**Comment by thelastinuit** [comment]
@philipstancil licensing?? dear zeus... hope not -_-

**Comment by gkovacs** [comment]
Windows isn't on the list of supported distributions because syslinux doesn't boot windows so it probably won't boot. If you're  trying to force it to install Windows to a NTFS USB drive nevertheless, use the command line option targetdrive and it may or may not work https://github.com/unetbootin/unetbootin/wiki/commands#targetdrive

**Comment by taniadaniela** [comment]
I'm having the getting stuck situation when trying to load an elementary linux iso while running unetbootin on mac os sierra 10.12.4.. I have tried two times and both times it get stuck and system monitor says it's not responding.

**Comment by hoxiea** [comment]
@taniadaniela Me too. I downloaded the new Ubuntu 17.04 .iso, just trying to use unetbootin to create a bootable jump drive on Mac OS X 10.12.4 per the [official Ubuntu install instructions](https://www.ubuntu.com/download/desktop/create-a-usb-stick-on-macos).

Following those instructions, I've fired up unetbootin three or four times now, and it freezes after extracting a few files. Spinning beach ball of death when the unetbootin window is active. Not Responding according to Activity Monitor. My most recent attempt has sat spinning for a few hours, so it's probably not just unresponsive while working on a big file.

Any thoughts, @gkovacs?

**Comment by harry-wood** [comment]
Just now I found it carried on running after leaving it for about half an hour (ubuntu iso, running unetbootin on mac os 10.9.5) 

...but that's really not obvious. I was assuming it was hung, and had previously tried to quit out of it and re-run, resulting in the even less obvious false success message.

**Comment by taniadaniela** [comment]
the really annoying of all this situation is that I went to a windows pc and got my iso burned on usb with unetbootin in less than 30 minutes. So it seems the problem is just around the mac version.

**Comment by hoxiea** [comment]
I never found a way to get unetbootin working on my 10.12.4 Mac. I tried letting unetbootin download the distro file, I tried a few different flavors of Ubuntu, nothing - it kept hanging exactly as I described above.

Eventually, I clicked the "Community Help Wiki" [link](https://help.ubuntu.com/community/How%20to%20install%20Ubuntu%20on%20MacBook%20using%20USB%20Stick) at the bottom of the [Official Ubuntu "Create a USB stick on macOS"](https://www.ubuntu.com/download/desktop/create-a-usb-stick-on-macos) and followed the "Manual Approach" instructions. It's actually quite easy to do, and once you've fired up `dd`, you can watch its progress in the "Disk" tab of Activity Monitor by looking at how many bytes its written. This worked for vanilla Ubuntu a couple of days ago, and I've got a Kubuntu `dd` running right now.

**Comment by gitmalong** [comment]
I have the same problem as @hoxiea . After some time I'm stuck with the spinning beach ball. I think the spinning beach ball issue is different then the lacking progress bar response issue when it is copying big files.

**Comment by antonpaisov** [comment]
same mac problem here

**Comment by qwangrepos** [comment]
same problem on Mac OS 10.11.6. However, the "manual approach" mentioned by @hoxiea worked pretty well for me. The "dd" step took quite a few minutes to finish.

**Comment by antonpaisov** [comment]
https://etcher.io/ worked for me :)

**Comment by spurdow** [comment]
yes I think the mac version is really unusable. I have the same problem I thought it was my laptop. The instructions on the ubuntu website should change it. This is base on my experience the unetbootin is really unusable in mac os.

**Comment by rgrosset** [comment]
Same problem here. Mac verison of Unetbootin freezes. Ubuntu 16.04 iso image.  Tried a number of different things no joy.

**Comment by ThomBMichigan** [comment]
I've got the same problem.  Are there any alternatives to Unetbootin?

**Comment by gitmalong** [comment]
@ThomBMichigan Ubuntus suggested [manual approach](https://help.ubuntu.com/community/How%20to%20install%20Ubuntu%20on%20MacBook%20using%20USB%20Stick) worked for me.

**Comment by Zack-od-Bins** [comment]
How to **check if UNetbootin is still copying** or not, when it appears to be frozen:

Copying filesystem.squashfs to my USB appeared to be stuck at 69% . Finally I found there is a way to check if the copying is still advancing, on Ubuntu:

Any application that can report the amount of free space on a device (or partition), like for example partitioning software GParted, can be used. In GParted I just had the disk (in this case the USB flashdrive) reread/refreshed. Every time I refreshed, I could see the amount of free space decreasing. Hence, it still was copying. 

It should probably work similarly on other operating systems.

**Comment by rondog** [comment]
I am using OSX 10.11.6 el capitan and mine was stuck on a squashfs file for a good 15 mins. I had just finished reading this thread and was going to give etcher a try and then it continued again and is now finished. It's like it knew I was about to give up on it

**Comment by dagolinuxoid** [comment]
This gist saved me from closing Unetbootin (its progress bar has been freezing on 13% for a while). Is you said — just wait patiently and all be ok :) PS. Elementary OS

**Comment by Xenero** [comment]
Ultimately it is a time based situation on this one.

I've had situations where one computer would take 40 minutes to copy and another takes only 4 or 5 minutes to copy that same ISO. Just have patience.

**Comment by six-h** [comment]
I'm struggling with Unetbootin, just tried to burn Mint 19 Cinnamon to a 2GB drive and after two hours, all I had was a copy of the ISO on the USB stick, ...not bootable!
Tried downloading Unetbootin from a different source and it appears the same (Surprise!) that has been running for about 6minutes and  seems to be running about the same, (3% in 6 minutes) estimate completion between 2 and 3 hours! yawn!
Wouldn't complain if it delivered what was expected after this time but to end up with just a duplicate of the ISO is more than a disappointment!
Grrr!

**Comment by qwertychouskie** [comment]
@six-h Did you use the latest version (https://github.com/unetbootin/unetbootin/releases/tag/661)?  You could also try https://etcher.io/

**Comment by six-h** [comment]
Thanks for replying, first was downloaded from Unertbootin's home page and file was labeled "unetbootin-windows-661".
Second one was downloaded from Sourceforge and file was labeled "unetbootin-windows-647"
Both took around two hours to produce nothing more than a copy of the ISO File on the target drive.

I had previously tried to burn a bootable copy from the same ISO File on a PC running a "Live DVD" of Linux Mint 17 32bit using the packaged  USB Formatting and Bootable USB drive burner apps.
Tried that three times, first hung at 20% second hung at 35% Third completed 100% but failed to verify.
each time I had the devils own job trying to recover the drive's full capacity (2.1GB) had to resort to Windoze Disk Management!!

Trying Etcher as you suggest, if not successful, maybe time to suspect the ISO Download,m ...or perish the thought, the 2GB stick.
If that's goosed, don't know where I could get any more of them, I find them handy, just enough for one OS or a few photos/videos without the worry of losing a lifetime's work!! 
 

S

**Comment by dylanh724** [comment]
Still having this issue in 2021. Swapped to Rufus and it works fine. There really needs to be a 2nd bar for giant files.

**Comment by theloneneutrino** [comment]
@philipstancil Windows has a separate tool for creating an installation media, is it possible that you could use that? I've tried creating a windows 10 installation media using unetbootin, and it fails to install windows even if the media creation works.

**Comment by dagolinuxoid** [comment]
@theloneneutrino recently I've installed windows 11 alongside my manjaro. In order to accomplish this I used *Ventoy*. As far as I'm concerned Unetbootin, Etcher and such are incapable to succeed at the task.
